cloud/ec2.go

```diff
@@ -767,35 +767,41 @@ func (m *ec2Manager) GetInstanceStatuses(ctx context.Context, hosts []host.Host)
 	return hostToStatusMap, nil
 }
 
-// GetInstanceStatus returns the current status of an EC2 instance.
-func (m *ec2Manager) GetInstanceStatus(ctx context.Context, h *host.Host) (CloudStatus, error) {
-	status := StatusUnknown
+// GetInstanceState returns a universal status code representing the state
+// of an ec2 and a state reason if available. The state reason should not be
+// used to determine the status of the ec2 but rather to provide additional
+// context about the state of the ec2 state.
+func (m *ec2Manager) GetInstanceState(ctx context.Context, h *host.Host) (CloudInstanceState, error) {
+	info := CloudInstanceState{Status: StatusUnknown}
 
 	if err := m.client.Create(ctx, m.region); err != nil {
-		return status, errors.Wrap(err, "creating client")
+		return info, errors.Wrap(err, "creating client")
 	}
 
 	instance, err := m.client.GetInstanceInfo(ctx, h.Id)
 	if err != nil {
 		if isEC2InstanceNotFound(err) {
-			return StatusNonExistent, nil
+			info.Status = StatusNonExistent
+			return info, nil
 		}
-		return status, err
+		return info, err
 	}
-	status = ec2StatusToEvergreenStatus(instance.State.Name)
 
-	if status == StatusRunning {
+	if info.Status = ec2StatusToEvergreenStatus(instance.State); info.Status == StatusRunning {
 		// Cache instance information so we can make fewer calls to AWS's API.
-		pair := hostInstancePair{
-			host:     h,
-			instance: instance,
-		}
-		if err = cacheAllHostData(ctx, m.env, m.client, pair); err != nil {
-			return status, errors.Wrapf(err, "caching EC2 host data for host '%s'", h.Id)
-		}
+		pair := hostInstancePair{host: h, instance: instance}
+		grip.Error(message.WrapError(cacheAllHostData(ctx, m.env, m.client, pair), message.Fields{
+			"message": "can't update host cached data",
+			"type":    "ec2",
+			"host_id": h.Id,
+		}))
```

Comment:
why is this being changed from returning an error to only logging an error?

Reply:
caching shouldn't be a requirement for this to pass I think, this was changed for cloud/ec2_fleet.go a while back but not changed for this one which I think was just an oversight